src/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -113,7 +148,13 @@ export default function VideoPlayer(
           helperText={helperText && showHelperInvalidText ? helperText : null}
           id={`${id}-componentWrapper`}
         >
-          <Box __css={styles.inside}>{embedCode}</Box>
+          <Box __css={styles.inside}>
+            {embedCodeFinal ? (
+              <span dangerouslySetInnerHTML={{ __html: embedCodeFinal }} />
+            ) : (
+              embedElement
+            )}
```

Comment:
For this, it might be better to move this up to the `embedElement` declaration so that you don't have to create the `iframe` element if you don't need it (e.g. if the `embedCodeFinal` is available). I don't think it's a big deal but if we don't need to create the element, it might be a small optimization.